src/hooks/useUploadsManager/useUploadsManager.tsx

```diff
@@ -161,6 +177,15 @@ export const UploadManagerProvider: React.FC = ({ children }) => {
           { leading: true }
         )
 
+        displaySnackbar({
```

Comment:
I don't understand what's happening here. It seems we're adding a separate snackbar for each upload and then use `useEffect` to filter them back? It doesn't seem like a good pattern, `UploadsManager` shouldn't be managing the snackbars that closely. It will now also rerender every time snackbars are modified (not really caring what the reason for that was).

I think that we should try to just throttle this on uploads manager end and not manage the snackbars at all. We could follow this policy: "When new upload starts, wait up to 500ms to see if other uploads start and count all of them. After 500ms pass without added upload, display a snackbar with number of uploads". This would be analogous for when asset finished uploading

Reply:
Let me know if you need more info on that